canvas_exercise/shapes_game/index.js

```diff
@@ -26,11 +14,166 @@ window.addEventListener("load", function() {
       seconds = 3,
       intervalId;
 
-  canvas.width = width;
-  canvas.height = height;
+      canvas.width = width;
+      canvas.height = height;
+
+  var shapeVal = 0;
+  var totalScore = 0;
+  var timerCountDown = 1;
+  var keyCodes = [37, 38, 39, 40];
+
+
+//-------------------------------------------------------------------  
+//Opening screen
+drawStartText();
+
+//-------------------------------------------------------------------  
+//Start Game
+  function startGame() {
+    clear(); //clear the canvas
+    drawRandomShape(); //add a random shape track scoring 
+    countDown(); //start timer countdown
+    endGaame() //use as a callback for when timer countdown expires  
+  }
+
+//-------------------------------------------------------------------
+
+function drawStartText() {
+  ctx.font = '40px serif';
+  ctx.fillStyle = 'orange';
+  ctx.fillText('Press the Space Bar to start a new game', 60, 340);
+}
+
+function drawStartTextAfterGame() {
+  ctx.font = '40px serif';
+  ctx.fillStyle = 'orange';
+  ctx.fillText('Game Over', 290, 340);  
+  ctx.fillText('Press the Space Bar to start a new game', 60, 380);  
+}
+
+//To start game, listen for spacebar keypress
+document.body.onkeypress = function(e){
+    if(e.keyCode === 32){
+      startGame();
+    } 
+}
+
+function clear() {
+  ctx.clearRect(0, 0, height, width); 
+}
+
+  function drawRandomShape() {
+    //draw random image
+    var randomImg = Math.floor(Math.random()*(5-1)+1);
+    switch(randomImg) {
+      case 1:
+        drawRedSquare()
+        break;
+      case 2:
+        drawWhiteSquare()
+        break;
+      case 3:
+        drawRedTriangle()
+        break;
+      case 4:
+        drawWhiteTriangle()
+    }
+    //compare image's value to keycode value, and calculate scoring
+    document.body.onkeydown = function(e){
+      if(e.keyCode == shapeVal) {
+        totalScore++
+        scoreSpan.innerHTML = totalScore;
+        clear()
+        drawRandomShape()
+      } else if(keyCodes.includes(e.keyCode)) {
+        totalScore--
+        scoreSpan.innerHTML = totalScore;
+        clear()
+        drawRandomShape()
+      }
+    }
+  }
+
+  function randomCoordinate() {
+    return Math.floor(Math.random()*(600-100)+100);
+  }
+
+//-------------------------------------------------
+
+function countDown() {
+    timerCountDown = 30;
+    var int = setInterval(function () {
```

Comment:
maybe call this something like intervalId.  Int is usually an integer type (which doesn't really exist in JS) so it can be a little confusing to name a variable int.